File: B/CPP/Tim.cpp
#include <string>
#include <iostream>
#include <vector>

using namespace std;

class Tim
{
private:
    string nama;
    string asalNegara;
    string tahunBerdiri;
    vector<string> pemain;

public:
    // constructor
    Tim() {}

    Tim(string nama, string asalNegara, string tahunBerdiri, vector<string> pemain)
    {
        this->nama = nama;
        this->asalNegara = asalNegara;
        this->tahunBerdiri = tahunBerdiri;
        this->pemain.insert(this->pemain.begin(), pemain.begin(), pemain.end());
    }

    // setter and getter
    void setNama(string nama)
    {
        this->nama = nama;
    }

    string getNama()
    {
        return this->nama;
    }

    void setAsalNegara(string asalNegara)
    {
        this->asalNegara = asalNegara;
    }

    string getAsalNegara()
    {
        return this->asalNegara;
    }

    void setTahunBerdiri(string tahunBerdiri)
    {
        this->tahunBerdiri = tahunBerdiri;
    }

    string getTahunBerdiri()
    {
        return this->tahunBerdiri;
    }

    void setPemain(vector<string> pemain)
    {
        this->pemain.insert(this->pemain.begin(), pemain.begin(), pemain.end());
    }

    vector<string> getPemain()
    {
        return this->pemain;
    }

    void addPemain(string pemain)
    {
        this->pemain.push_back(pemain);
    }

    // output
    void printTim()
    {
        cout << "Nama Tim       : " << this->getNama() << endl;
        cout << "Asal Negara    : " << this->getAsalNegara() << endl;
        cout << "Tahun Berdiri  : " << this->getTahunBerdiri() << endl;
        cout << "Pemain         : " << endl;
        for (auto const &c : this->getPemain())
        {
            cout << "-" + c << endl;
        }
    }

    ~Tim() {}
};

File: B/Java/Tim.java
import java.util.Arrays;

public class Tim {
    private String nama;
    private String asalNegara;
    private String tahunBerdiri;
    private String[] pemain;

    // constructor
    public Tim() {
    }

    public Tim(String nama, String asalNegara, String tahunBerdiri, String[] pemain) {
        this.nama = nama;
        this.asalNegara = asalNegara;
        this.tahunBerdiri = tahunBerdiri;
        this.pemain = pemain;
    }

    // setter and getter method
    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getNama() {
        return this.nama;
    }

    public void setAsalNegara(String asalNegara) {
        this.asalNegara = asalNegara;
    }

    public String getAsalNegara() {
        return this.asalNegara;
    }

    public void setTahunBerdiri(String tahunBerdiri) {
        this.tahunBerdiri = tahunBerdiri;
    }

    public String getTahunBerdiri() {
        return this.tahunBerdiri;
    }

    public void setPemain(String[] pemain) {
        this.pemain = pemain;
    }

    public String[] getPemain() {
        return this.pemain;
    }

    public void addPemain(String pemain) {
        this.pemain = Arrays.copyOf(this.pemain, this.pemain.length + 1);
        this.pemain[this.pemain.length - 1] = pemain;
    }

    // output
    public void printTim() {
        System.out.println("Nama Tim        : " + this.getNama());
        System.out.println("Asal Negara     : " + this.getAsalNegara());
        System.out.println("Tahun Berdiri   : " + this.getTahunBerdiri());
        System.out.println("Pemain          : ");
        for (String p : this.getPemain()) {
            System.out.println("- " + p);
        }
    }
}
